arbos/storage/storage_test.go

```diff
@@ -91,3 +96,73 @@ func TestStorageBackedBigInt(t *testing.T) {
 		})
 	}
 }
+
+func TestOpenCachedSubStorage(t *testing.T) {
+	s := NewMemoryBacked(burn.NewSystemBurner(nil, false))
+	var subSpaceIDs [][]byte
+	for i := 0; i < 20; i++ {
+		subSpaceIDs = append(subSpaceIDs, []byte{byte(rand.Intn(0xff))})
+	}
+	var expectedKeys [][]byte
+	for _, subSpaceID := range subSpaceIDs {
+		expectedKeys = append(expectedKeys, crypto.Keccak256(s.storageKey, subSpaceID))
+	}
+	n := len(subSpaceIDs) * 50
+	start := make(chan struct{})
+	errs := make(chan error, n)
+	var wg sync.WaitGroup
+	for i := 0; i < n; i++ {
+		j := i % len(subSpaceIDs)
+		subSpaceID, expectedKey := subSpaceIDs[j], expectedKeys[j]
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			<-start
+			ss := s.OpenCachedSubStorage(subSpaceID)
+			if !bytes.Equal(ss.storageKey, expectedKey) {
+				errs <- fmt.Errorf("unexpected storage key, want: %v, have: %v", expectedKey, ss.storageKey)
+			}
+		}()
+	}
+	close(start)
+	wg.Wait()
+	select {
+	case err := <-errs:
+		t.Fatal(err)
+	default:
+	}
```

Comment:
For reporting mismatch you should opt for Errorf [0]. One would use Fatalf more fore something that makes rest of the test meaningless, e.g. error when setting up RPC connection or something like that.

And we should display all errors from that channel.

[0] https://google.github.io/styleguide/go/decisions#keep-going